Small documentation fixups (#26999)
cilium-team/cilium: Documentation/network/servicemesh/mutual-authentication/installation.rst

.. tabs::

    .. group-tab:: Cilium CLI

        .. include:: ../../../installation/cli-download.rst

>         Mutual authentication and associated SPIRE server can be enabled with the following command.
>         Note this requires the Cilium CLI Helm mode (version 0.15 or later).

Reviewer: Use active voice, specific is better than general, punctuation nits

I'm not attached to "its" re:SPIRE server, feel free to replace with another article like "an"
```suggestion
        You can enable mutual authentication and its associated SPIRE server with the following command.
        This command requires the Cilium CLI Helm mode version 0.15 or later.
```

        .. code-block:: shell-session

            $ cilium install \
                --helm-set authentication.mutual.spire.enabled=true \
                --helm-set authentication.mutual.spire.install.enabled=true

>         Next you can check the status of the Cilium agent and operator:

Reviewer: Good use of "you" to address the user, but we can trim it here
```suggestion
        Next, check the status of the Cilium agent and operator:
```
